web/src/components/Navigation.jsx

```diff
@@ -344,16 +353,46 @@ const SubscriptionItem = (props) => {
   );
 };
 
-const NotificationGrantAlert = (props) => {
+const NotificationPermissionRequired = ({ refreshPermissions }) => {
+  const { t } = useTranslation();
+  return (
+    <Alert severity="info" sx={{ paddingTop: 2 }}>
+      <AlertTitle>{t("alert_notification_permission_required_title")}</AlertTitle>
+      <Typography gutterBottom align="left">
+        {/* component=Button is not an anchor, false positive */}
+        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+        <Link
+          component="button"
+          style={{ textAlign: "left" }}
+          onClick={async () => {
+            await notifier.maybeRequestPermission();
+            refreshPermissions();
+          }}
+        >
+          {t("alert_notification_permission_required_description")}
+        </Link>
+      </Typography>
+    </Alert>
+  );
+};
+
+const NotificationPermissionDeniedAlert = () => {
+  const { t } = useTranslation();
+  return (
+    <Alert severity="warning" sx={{ paddingTop: 2 }}>
+      <AlertTitle>{t("alert_notification_permission_denied_title")}</AlertTitle>
+      <Typography gutterBottom>{t("alert_notification_permission_denied_description")}</Typography>
+    </Alert>
+  );
+};
+
+const NotificationIOSInstallRequiredAlert = () => {
   const { t } = useTranslation();
   return (
     <>
-      <Alert severity="warning" sx={{ paddingTop: 2 }}>
-        <AlertTitle>{t("alert_grant_title")}</AlertTitle>
-        <Typography gutterBottom>{t("alert_grant_description")}</Typography>
-        <Button sx={{ float: "right" }} color="inherit" size="small" onClick={props.onRequestPermissionClick}>
-          {t("alert_grant_button")}
-        </Button>
+      <Alert severity="info" sx={{ paddingTop: 2 }}>
+        <AlertTitle>{t("alert_notification_ios_install_required_title")}</AlertTitle>
+        <Typography gutterBottom>{t("alert_notification_ios_install_required_description")}</Typography>
       </Alert>
```

Comment:
If I understand correctly, this will show up for all iOS devices and it will persist unless you do what the description says. What if somebody just wants to use the ntfy web app with web sockets and tab open? Will this alert stay there? 

If this is the case, we should add a [Dismiss] button, right?

Reply:
That's correct, but it's similar to the current implementation, it shows a persistent "not supported" on iOS:

![15B87DDD-7702-466F-A18B-85E014055968_1_105_c](https://github.com/binwiederhier/ntfy/assets/132819643/6abacbb7-7a17-4e56-a920-271f35415e1f)

So this change would basically change it to something the user can fix if wanted.